### PR TITLE
Add rancher component modules

### PR DESCRIPTION
**Pull request by git-ival**
#### This PR includes:
* A refactor of the `rancher-cluster-operations` directory into a collection of convenient component modules that can be applied separately by passing in the proper input variables
* An update to the main README.md to include details on each sub-directory's usage and  additional recommended dev tools
* Added documentation and examples for `rancher-cluster-operations` modules

#### Testing
Apply each module against a running rancher control-plane. An example tfvars is not provided because this is a collection of modules with a variety of input variables which are documented in each convenience module's README.md (located in the convenience module's directory).

The `rancher-cloud-credential`, `rancher-node-template` and `rancher-cluster` (v1) modules can be tested by following the basic testing steps in PR #56 .

Notes: 
* The `install-common` module is identical to the `install-common` child module included in the `control-plane` root module, so this should not need testing
* The `control-plane`'s `install-common` child module is replaced by this component module
* The `rancher-controller-metrics` module should not be used if running rancher 2.6.x with the latest rancher-monitoring installed

### CODE REVIEW

**Review by floatingman — COMMENTED** [review]
Just a few comments. 

Also do you think there should be some examples on how someone might use the modules?

File: rancher-cluster-operations/rancher-cloud-credential/variables.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  validation {
+    condition     = contains(["aws", "linode"], var.cloud_provider)
+    error_message = "Please pass in a case-sensitive string equal to one of the following: [\"aws\", \"linode\"]."
+  }
+}
+
+variable "credential_config" {
+  type      = any
+  nullable  = false
+  sensitive = true
```
[the remark this answers is not in the document]
Comment (git-ival) [reply rancher-cluster-operations/rancher-cloud-credential/variables.tf:24]: I will add descriptions to each variable for clarifications but I think the expectation of what should be passed in is clear based on the usage of `var.credential_config` in `main.tf`.

I will also add an example for this module just to be sure. Below are two brief examples for now.

For a linode credential:
```
module "cloud_credential" {
  ...
  cloud_provider = "linode"
  credential_config = {
    token = <a linode API token>
  }
}
```

For an AWS credential:
```
module "cloud_credential" {
  ...
  cloud_provider = "aws"
  credential_config = {
    access_key     = <an aws access key>
    secret_key     = <an aws secret key>
    default_region = <the aws region for this credential>  
    }
}
```

**Review by floatingman — COMMENTED** [review]
Just a few questions.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,80 @@
...
+
+## Recommended Dev Tools
+- [pre-commit](https://pre-commit.com/#install)
+  - [pre-commit-hooks](https://github.com/pre-commit/pre-commit-hooks)
+  - [pre-commit-terraform](https://github.com/antonbabenko/pre-commit-terraform)
+---
+### Setup Recommended pre-commit Hooks
+1. Install `pre-commit`
+2. Create `pre-commit-config.yaml` in the git working directory of this repository
+3. Add `pre-commit-hooks` to your `pre-commit-config.yaml`
```
[the remark this answers is not in the document]
Comment (git-ival) [reply README.md:27]: I am still tinkering with pre-commit and tflint and didn't want to further bloat the PR with these files.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,80 @@
...
+            - id: terraform_fmt
+                args:
+                - --args=-recursive
+            - id: terraform_docs
+                args:
+                - --hook-config=--path-to-file=README.md # Valid UNIX path. I.e. ../TFDOC.md or docs/README.md etc.
+                - --hook-config=--add-to-existing-file=true # Boolean. true or false
+                - --hook-config=--create-file-if-not-exist=true # Boolean. true or false
+        ```
+5. Create a `.tflint.hcl` configuration in the git working directory of this repository
```
[the remark this answers is not in the document]
Comment (git-ival) [reply README.md:62]: I am still tinkering with pre-commit and tflint and didn't want to further bloat the PR with these files.

**Pull request merged**